src/libsyncengine/jobs/network/API_v2/uploadjob.cpp

```diff
@@ -184,8 +184,35 @@ std::string UploadJob::getContentType(bool &canceled) {
 }
 
 ExitInfo UploadJob::readFile() {
-    std::ifstream file(_filePath, std::ios_base::in | std::ios_base::binary);
-    if (!file.is_open()) {
+    std::ifstream file;
+    int count = 0;
+    do {
+        file.open(_filePath, std::ios_base::in | std::ios_base::binary);
+        if (!file.is_open()) {
+            bool exists = false;
+            IoError ioError = IoError::Success;
+            if (!IoHelper::checkIfPathExists(_filePath, exists, ioError)) {
+                LOGW_WARN(_logger, L"Error in IoHelper::checkIfPathExists: " << Utility::formatIoError(_filePath, ioError));
+                return {ExitCode::SystemError, ExitCause::Unknown};
+            }
+            if (ioError == IoError::AccessDenied) {
+                LOGW_WARN(_logger, L"Access denied to " << Utility::formatSyncPath(_filePath));
+                return {ExitCode::SystemError, ExitCause::FileAccessError};
+            }
+            if (!exists) {
+                LOGW_DEBUG(_logger, L"Item does not exist anymore - " << Utility::formatSyncPath(_filePath));
+                return {ExitCode::SystemError, ExitCause::NotFound};
+            }
+            Utility::msleep(1000);
+        }
+
+        // Some applications generate locked temporary files during save operations. To avoid spurious "access denied" errors,
+        // we retry for 10 seconds, which is usually sufficient for the application to delete the tmp file. If the file is still
+        // locked after 10 seconds, a file access error is displayed to the user. Proper handling is also implemented for
+        // "file not found" errors.
+    } while (count++ < 10 && !file.is_open());
+
+    if (count >= 10) {
```

Comment:
Each unit test involving this function will last at least 10 seconds when it returns false.

**Suggestion.** Extract a method for which `count` is an argument. Test it independently. For the full test, you can mock this method by means of inheritance. Alternatively, make `count` a member variable that can be customized for unit tests.

Reply:
https://github.com/Infomaniak/desktop-kDrive/pull/396#discussion_r1847964890